### ISSUE TITLE
Can't use custom font with image in one document

### ISSUE THREAD

**Issue by WebKieth** [diegomura/react-pdf#967]
**Describe the bug**
I'm working on russian language interface and need to render document from form with some text info and barcode. So I need russian langage symbols and barcode image in one doc, which I need to render on the fly. 
When I register custom cyrillic font I always get unexpected error:
`Cannot read property 'hasGlyphForCodePoint' of null`
And no differ how to register: I try from server, from public folder and sets import. Error always thrown.
About image:
Firstly, docs wrote wrong, {data: Buffer, format: ''} wont render image - I try to pass buffer from canvas (initially - barcode has svg format and rendered from number by react-barcode). I could render it only when read in one of your issues about {uri: '', format: ''} and passing base64 uri instead of affay buffer and image starts working.
Interesting moment: 
`Cannot read property 'hasGlyphForCodePoint' of null` don't broke text - fonts are render, but image - isn't... magic?..
But when I remove `Font.register` fonts aren't render, but image - is. Error still there but different: `Error: stream.push() after EOF`
🤯 🤬
**To Reproduce**
Form:
```
@observer
class TransportInvoiceForm extends React.Component {
    constructor(props) {
        super(props)
        this.barcodeElement = React.createRef()
    }

    @observable barcodeImage;
    @action createBarcodeImage() {
        const el = this.barcodeElement.current;
        if (!el.refs || !el.refs.renderElement) return;
        const svg = el.refs.renderElement,
        {width, height} = svg.getBBox(),
        clone = svg.cloneNode(true),
        xml = clone.outerHTML,
        blob = new Blob([xml],{type:'image/svg+xml;charset=utf-8'}),
        URL = window.URL || window.webkitURL || window,
        blobURL = URL.createObjectURL(blob),
        image = new Image();
        image.onload = () => {
            const canvas = document.createElement('canvas');
            canvas.widht = width;
            canvas.height = height;
            const context = canvas.getContext('2d');
            context.drawImage(image, 0, 0, width, height );
            let jpeg = canvas.toDataURL();
            this.barcodeImage = jpeg
        }
        image.src = blobURL;
    }

    // unimportant observable-action pairs and update hooks....
 
    render() {
        return <React.Fragment>
            {opened ? <Overlay>
                <div className='w-full flex flex-col relative h-full'>
                    // unimportant form code there....
                    <div className='self-end w-full foot-panel h-64px'>
                        <div className='flex justify-between'>
                            <PDFDownloadLink document={
                                    <TranspostInvoiceDocument driverInfo={this.driver} vehicleInfo={this.vehicle}
                                        barcodeImage={this.barcodeImage}/>
                                }
                                fileName={`ttn-${postingNumber}.pdf`}>
                                {({loading}) => {
                                    return <Button secondary className='mr-4'>
                                        {loading ? <Loader className='mr-2' sm overlayed={false}/> : <Icon className='stroke-blue mr-2' name='thin--doc'/>}
                                        <Trans i18nKey='common.actions.savePDF'/>
                                    </Button>
                                }}
                            </PDFDownloadLink>
                        </div>
                    </div>
                </div>
            </Overlay> : ''}
        </React.Fragment>
    }
}
```

Document: 
```
import React from 'react'
import { Document, Page, View, Text, StyleSheet, Font, Image } from '@react-pdf/renderer'
import {Trans} from 'react-i18next'
import source from '@/assets/Roboto.ttf'

Font.register({ family: 'Roboto', src: source, format: "truetype", });

const transportInvoiceStyles = StyleSheet.create({
    page: {
        fontFamily: 'Roboto',
        flexDirection: 'column',
        margin: 20
    },
    title: {
        fontSize: 24,
        marginBottom: 20,
        // fontFamily: 'Roboto'
    },
    section: {
        marginBottom: 10,
        flexDirection: 'column'
    },
    field: {
        width: '80%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        fontSize: 14,
        // fontFamily: 'Roboto'
    },
    label: {
        marginBottom: 10
    }
})

class TranspostInvoiceDocument extends React.Component {
    render() {
        const {driverInfo, vehicleInfo, barcodeImage} = this.props 
        return <Document>
            <Page size='A4' style={transportInvoiceStyles.page}>
                <Text style={transportInvoiceStyles.title}><Trans i18nKey='pdf.posting.transportinvoice.title'/></Text>
                  // unimportant text markup
                <View style={transportInvoiceStyles.section}>
                    // unimportant text markup
                    <Image source={{uri: barcodeImage, format: 'png'}}/>
                </View>
            </Page>
        </Document>
    }
}


export {TranspostInvoiceDocument}
```

**Expected behavior**
Font wont throw error on register, image should render like docs says - from buffer without errors

**Screenshots**
<img width="1207" alt="Снимок экрана 2020-08-06 в 02 22 03" src="https://user-images.githubusercontent.com/12966747/89473680-2d158600-d78c-11ea-84ee-3e1dcad9ff5c.png">
<img width="836" alt="Снимок экрана 2020-08-06 в 02 22 22" src="https://user-images.githubusercontent.com/12966747/89473684-2edf4980-d78c-11ea-9659-6f860a40415e.png">
<img width="836" alt="Снимок экрана 2020-08-06 в 02 23 04" src="https://user-images.githubusercontent.com/12966747/89473686-2f77e000-d78c-11ea-88ea-b7808a868b48.png">

**Desktop (please complete the following information):**
 - OS: latest OSX Catalina
 - Browser latest chrome
 - React-pdf version 1.6.10, tries 2.0.0-beta.13 renderer too

six hours of analysis and research. I think it should be really simpler to write canvas + jsPDF external module, instead of your 🐞 solution

**Comment by WebKieth** [comment]
I solve my working task by rendering fonts inside canvas and converting this into image/png base64, which sets to <Image/> prop. Font quality is bad, but for first time it's enough.
Still get `Error: stream.push() after EOF`. I saw some issues right about it...

<img width="1111" alt="Снимок экрана 2020-08-06 в 03 49 36" src="https://user-images.githubusercontent.com/12966747/89478151-dca42580-d797-11ea-9001-441607ceb1b4.png">

**Comment by diegomura** [comment]
This should be fixed in latest 2.0 version! Please check and report back if not 😄 Thanks!
